File: backendworker/NHSOnline.Backend.Worker/OrganDonation/Models/LookupRegistrationRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;

namespace NHSOnline.Backend.Worker.OrganDonation.Models
{
    public class LookupRegistrationRequest
    {
        [IgnoreDataMember]
        [Required]
        public string NhsNumber { get; set; }

        [Required]
        public string Identifier => $"https://fhir.nhs.uk/Id/nhs-number|{NhsNumber}";

        [Required]
        public string Given { get; set; }

        [Required]
        public string Family { get; set; }

        [Required]
        public string BirthDate { get; set; }
    }
}

File: bddtests/src/main/kotlin/mocking/organDonation/models/LookupRegistrationRequest.kt
package mocking.organDonation.models

import models.Patient

data class LookupRegistrationRequest(
        var identifier: String,
        var given: String,
        var family: String,
        var birthdate: String
)
{
    companion object {
        fun forPatient(patient: Patient): LookupRegistrationRequest {
            val nhsNumber = patient.nhsNumbers.first()
            return LookupRegistrationRequest(
                    identifier = "https://fhir.nhs.uk/Id/nhs-number|"
                            + nhsNumber,
                    given = patient.firstName,
                    family = patient.surname,
                    birthdate = patient.dateOfBirth
            )
        }
    }
}
